### PR TITLE
MM-19663 | Migrate brand_test and cluster_test to testify

### LINKED ISSUE #12890: Migrate tests from "api4/cluster_test.go" and "api4/brand_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `api4/cluster_test.go` and `api4/brand_test.go` files.

The expected way to migrate this is to go to the `api4/cluster_test.go` and `api4/brand_test.go` files and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in [https://community.mattermost.com/](https://community.mattermost.com/)

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19663

### PR DESCRIPTION
**Pull request by m4ver1k**
#### Summary

Migrate the tests in the api4/cluster_test.go and api4/brand_test.go to use testify

#### Ticket Link

[MM-19663](https://mattermost.atlassian.net/browse/MM-19663)
Fixes #12890

### CODE REVIEW

**Comment by m4ver1k** [comment]
This PR Fixes #12890

File: api4/cluster_test.go [hunk 1]
```diff
@@ -22,9 +23,8 @@ func TestGetClusterStatus(t *testing.T) {
 		infos, resp := th.SystemAdminClient.GetClusterStatus()
 		CheckNoError(t, resp)
 
-		if infos == nil {
-			t.Fatal("should not be nil")
-		}
+		require.NotNil(t, infos, "cluster status should not be nil")
+
```
Comment (hanzei) [inline api4/cluster_test.go]: Please remove this empty line.

**Comment by m4ver1k** [comment]
I will fox these and update PR

File: api4/cluster_test.go (hunk 1 above)
Comment (m4ver1k) [reply api4/cluster_test.go]: removed :)

File: api4/brand_test.go [hunk 2]
```diff
@@ -48,7 +47,7 @@ func TestUploadBrandImage(t *testing.T) {
 	} else if resp.StatusCode == http.StatusUnauthorized {
 		CheckUnauthorizedStatus(t, resp)
 	} else {
-		t.Fatal("Should have failed either forbidden or unauthorized")
+		require.FailNow(t, "Should have failed either forbidden or unauthorized")
```
Comment (hanzei) [inline api4/brand_test.go]: `FailNow` ends the execution of also all other tests, which is not ideal. 
```suggestion
		require.Fail(t, "Should have failed either forbidden or unauthorized")
```

Comment (m4ver1k) [reply api4/brand_test.go]: OK thanks for the feedback,
I was replacing it with something similar to `t.Fatal` which also calls a `FailNow`
Will update this right away

**Comment by m4ver1k** [comment]
@hanzei `Setup` in the CI-PR-Merge pipeline failed with:
`Caused by: org.jenkinsci.plugins.workflow.steps.MissingContextVariableException: Required context class hudson.FilePath is missing`

**Comment by reflog** [comment]
@m4ver1k - I restarted Travis task. All good, thank you!

**Pull request merged**